PROTO_tests/tests/random_test.py

```diff
@@ -47,70 +53,81 @@ def test_integers(self):
-        rng = ak.random.default_rng(18)
+    @pytest.mark.parametrize("data_type", INT_FLOAT)
+    def test_shuffle(self, data_type):
 
-        int_pda = rng.integers(-(2**32), 2**32, 10)
-        pda_copy = int_pda[:]
-        # shuffle int_pda in place
-        rng.shuffle(int_pda)
-        # verify all the same elements are in permutation as the original
-        assert (ak.sort(int_pda) == ak.sort(pda_copy)).all()
+        # ints are checked for equality; floats are checked for closeness
 
-        float_pda = rng.uniform(-(2**32), 2**32, 10)
-        pda_copy = float_pda[:]
-        rng.shuffle(float_pda)
-        # verify all the same elements are in permutation as the original
-        assert (ak.sort(float_pda) == ak.sort(pda_copy)).all()
+        check = lambda a, b, t: (
+            (a == b).all() if t is ak.int64 else np.allclose(a.to_list(), b.to_list())
+        )
 
-        rng = ak.random.default_rng(18)
+        # verify all the same elements are in the shuffle as in the original
 
-        pda = rng.integers(-(2**32), 2**32, 10)
+        rng = ak.random.default_rng(18)
+        rnfunc = rng.integers if data_type is ak.int64 else rng.uniform
+        pda = rnfunc(-(2**32), 2**32, 10)
+        pda_copy = pda[:]
         rng.shuffle(pda)
-        assert (pda == int_pda).all()
 
-        pda = rng.uniform(-(2**32), 2**32, 10)
-        rng.shuffle(pda)
-        assert np.allclose(pda.to_list(), float_pda.to_list())
+        assert check(ak.sort(pda), ak.sort(pda_copy), data_type)
 
-    def test_permutation(self):
         # verify same seed gives reproducible arrays
+
+        rng = ak.random.default_rng(18)
+        rnfunc = rng.integers if data_type is ak.int64 else rng.uniform
+        pda_prime = rnfunc(-(2**32), 2**32, 10)
+        rng.shuffle(pda_prime)
+
+        assert check(pda, pda_prime, data_type)
+
+    @pytest.mark.parametrize("data_type", INT_FLOAT)
+    def test_permutation(self, data_type):
+
+        # ints are checked for equality; floats are checked for closeness
+
+        check = lambda a, b, t: (
+            (a == b).all() if t is ak.int64 else np.allclose(a.to_list(), b.to_list())
+        )
+
+        # verify all the same elements are in the permutation as in the original
+
         rng = ak.random.default_rng(18)
-        # providing just a number permutes the range(num)
         range_permute = rng.permutation(20)
-        assert (ak.arange(20) == ak.sort(range_permute)).all()
+        assert (ak.arange(20) == ak.sort(range_permute)).all()  # range is always int
 
-        pda = rng.integers(-(2**32), 2**32, 10)
-        array_permute = rng.permutation(pda)
-        # verify all the same elements are in permutation as the original
-        assert (ak.sort(pda) == ak.sort(array_permute)).all()
+        # verify same seed gives reproducible arrays
+
+        rng = ak.random.default_rng(18)
+        rnfunc = rng.integers if data_type is ak.int64 else rng.uniform
+        pda = rnfunc(-(2**32), 2**32, 10)
```

Comment:
this is nuanced but doing parameterization here is not testing quite the same thing as the original. Since this is only using the same rng for a single function call but the original is making sure that subsequent calls to a generator are also consistent. But it's not actually making sure the second call differs form the first... maybe this isn't that important

I think I need to chat about this with the group since I'm struggling to decide if this slight difference actually matters or if we should create a separate test for this kinda thing

Reply:
Oh, I see it.  Without parameterization, one rng is created that both INT and FLOAT use.  With parameterization, each gets its own.  It's true  that that's a difference, but I can't see how it's important.

Reply:
covered in team sync I think this isnt an issue for shuffle / permutation, but I think adding another test to make sure we capture this duplicated calls to a generator with different dtypes and make sure the state is carried over